share/scripts/breakpoints.tcl

```diff
@@ -0,0 +1,41 @@
+# Easily save breakpoints to and restore from a file
+
+proc save_breakpoints_help {args} {
+	return {save_breakpoints [<filename>]
+
+Save all breakpoints to a given file.
+}}
+
+proc save_breakpoints {filename} {
+	set all [debug breakpoint list]
+	set fh [open $filename "w"]
+	puts $fh [list {*}$all]
+	close $fh
+}
+set_help_proc save_breakpoints save_breakpoints_help
+
+proc load_breakpoints_help {args} {
+	return {load_breakpoints [<filename>]
+
+Load only new breakpoints from a given file, while preserving pre-existing ones if any.
+}}
+
+proc load_breakpoints {filename} {
+	set fh [open $filename "r"]
+	set all [gets $fh]
+	close $fh
+	# compare and create breakpoints if not found
+	foreach {_ data1} [eval list $all] {
+		set found 0
+		foreach {_ data2} [debug breakpoint list] {
+			if {$data1 eq $data2} {
+				set found 1
+			}
+		}
```

Comment:
I only thought about this now ... this implementation is correct, but might be slow when there are many breakpoints, it's an O(n^2) algorithm.

The following suggestion is still O(n^2), but probably (i haven't actually measured) already a lot faster because the inner loops is now moved from the Tcl script to a builtin Tcl function (implemented in C).

```
set existing_bps [dict values [debug breakpoint list]]
foreach {_ bp} [eval list $all] {
    if ($bp in $existing_bps} continue
    ...
}
```
The inner loop is now hidden inside the `in` operator.